### DISCUSSION TITLE
How to allow user configurable light and dark colours to be bindable

### DISCUSSION THREAD

**Discussion by JohnHDev** [dotnet/maui#16305] [Q&A] [answered]
We have a number of ways to set light and dark colours for a control, but I can't see a way to make those colours bindable. So we have a situation where we have to refresh the entire UI after a user has made a colour config change.

Am I missing something or is this just not possible with Maui? We implemented dark mode in Xamarin Forms before Microsoft did, and we built that to work with multi binding and bindable properties for those colours, plus property converters. You know, how developers would want to use it.
None of that seems possible with Maui unless we roll my own, again.

**Comment by JRosanowski** [comment]
Have you tried AppThemeBinding?
https://learn.microsoft.com/en-us/dotnet/maui/user-interface/system-theme-changes

**Reply by JohnHDev** [reply]
AppThemeBinding doesn't support dynamic light and dark colours, e.g. if the user changes what light or dark colour to use at run time. Unless I have missed something? I ideally want to bind the light and dark colours of AppThemBinding to values in the view model.

The use case here is the user is running the app, goes to the settings page and changes the app background colours used for both light and dark mode. I don't want to have to regenerate the UI, I want to just update the VM with the new values.
This isn't limited to just background colours of course, our XF app has many user configurable assets.

**Comment by developer9969** [comment, marked as answer]
Not sure if this is what you are looking for, we do not use AppThemebinding at all we have many themes if you like 
when a user switches theme we remove the current theme(ResourceDictionary) and we load the selected theme (another resource dictionary)

so the approached used here is that given a set of themes the user chooses a theme and the app changes the color scheme . and we have some pages where we bind a color to a control and the user can change the color too but always bound to a resource dictionary where the colors are defined

**Reply by JohnHDev** [reply]
That sounds exactly what Im after, are you able to share the code for removing existing and adding another theme to the ResourceDictionary? Is it the app Resources you are updating or for individual pages?

Many thanks!

**Reply by JohnHDev** [reply]
Actually, using merged resource dictionaries was the way to go and I can now cleanly swap out new user defined light and dark colours dynamically. @developer9969, I salute you. :)

**Comment by developer9969** [comment]
@JohnHDev  glad it helped we have used it successfully in both XF and Maui in switching between LightTheme vs DarkTheme all driven by swapping resource dictionaries
